src/selectors/pipeline.test.js

```diff
@@ -0,0 +1,120 @@
+      const activePipeline = undefined;
+      const newMockState = reducer(
+        mockState.animals,
+        updateActivePipeline(activePipeline)
+      );
+      const nodeDisabledPipeline = getNodeDisabledPipeline(newMockState);
+      expect(
+        mockState.animals.node.ids.every(
+          nodeID => !nodeDisabledPipeline[nodeID]
+        )
+      ).toBe(true);
+    });
+
+    it('does not disable any nodes that are in the active pipeline', () => {
+      const activePipeline = 'ds';
+      const activePipelineNodeIDs = mockState.animals.node.ids.filter(
+        nodeID => mockState.animals.node.pipelines[nodeID][activePipeline]
+      );
+      const newMockState = reducer(
+        mockState.animals,
+        updateActivePipeline(activePipeline)
+      );
+      const nodeDisabledPipeline = getNodeDisabledPipeline(newMockState);
+      expect(
+        activePipelineNodeIDs.every(nodeID => !nodeDisabledPipeline[nodeID])
+      ).toBe(true);
+    });
+
+    it('disables every node that is not in the active pipeline', () => {
+      const activePipeline = 'de';
+      const inactivePipelineNodeIDs = mockState.animals.node.ids.filter(
+        nodeID => !mockState.animals.node.pipelines[nodeID][activePipeline]
+      );
+      const newMockState = reducer(
+        mockState.animals,
+        updateActivePipeline(activePipeline)
+      );
+      const nodeDisabledPipeline = getNodeDisabledPipeline(newMockState);
+      expect(
+        inactivePipelineNodeIDs.every(nodeID => nodeDisabledPipeline[nodeID])
+      ).toBe(true);
+    });
+  });
+
+  describe('getPipelineNodeIDs', () => {
+    const nodeIDs = getNodeIDs(mockState.animals);
+    const pipelineNodeIDs = getPipelineNodeIDs(mockState.animals);
+    const nodePipelines = getNodePipelines(mockState.animals);
+    const activePipeline = getActivePipeline(mockState.animals);
+
+    it('returns an array of node IDs', () => {
+      expect(pipelineNodeIDs).toEqual(expect.arrayContaining(nodeIDs));
+    });
+
+    it('does not contain any nodes that are not in the current pipeline', () => {
+      expect(
+        pipelineNodeIDs.every(nodeID => nodePipelines[nodeID][activePipeline])
+      ).toBe(true);
+    });
+
+    it('contains all nodes in the current pipeline', () => {
+      const inactivePipelineNodeIDs = pipelineNodeIDs.filter(
+        nodeID => nodePipelines[nodeID][activePipeline]
+      );
+      inactivePipelineNodeIDs.forEach(nodeID => {
+        expect(pipelineNodeIDs).toContain(nodeID);
+      });
+    });
+  });
+
+  describe('getPipelineTagIDs', () => {
+    it('returns an array of tag IDs', () => {
+      expect(getPipelineTagIDs(mockState.animals)).toEqual(
+        expect.arrayContaining(getTagIDs(mockState.animals))
+      );
+    });
+
+    it('does not contain any tags that are not in the current pipeline', () => {
+      // Tag 'huge' is not in pipeline 'de'
+      const activePipeline = 'de';
```

Comment:
Seems like this test could be too data set specific and therefore unreliable? Ideally the set up for it would be isolated in here.

Reply:
Yes I agree, but I couldn't think of a good way of doing it while essentially replicating the selector. unless.. hmm, maybe I could programmatically add an extra tag/node/pipeline without manually adding them to the dataset?

Reply:
Sounds fine to me, I think it would be a good idea as currently the test is not going to be as robust.